File: projectEuler/26.py
def div(a,b):
	if (a%b==0):
		return True
	else:
		return False

def cnt(x):
	y = int(x)
	while y%2==0:
		y= y/2
	while y%5==0:
		y = y/5
	a = 9
	ans = 1
	while True:
		if div(a,y):
			break
		a = 10*a
		a = a + 9
		ans = ans +1
	return ans
entrada = input()
arr = entrada.split(' ')
for i in range(len(arr)):
	valor = cnt(arr[i])
	print(arr[i],end=" ")
	print(valor)

File: projectEuler/26.cpp
#include<bits/stdc++.h>

using namespace std;
typedef long long ll;

ll M[1005];

ll f(int x){
	ll ans = 1;
	for(int i=2;i*i<=x;i++){
		int j = 1;
		while(x%i==0){
			j*=i;
			x/=i;
		}
		ans *= M[j];
	}
	ans *= M[x];
	return ans;
}

int main(){
	int a,b;
	while(cin>>a>>b){
		M[a] = b;
	}
	int ans = -1;
	ll valor = 0;
	for(int i=1;i<1000;i++){
		ll val = f(i);
		if(val > valor){
			valor = val;
			ans = i;
		}
	}
	cout<<ans<<" "<<valor<<endl;

	return 0;
}

